tests/08-native-runs/14-smallprng/test-smallprng.c

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright (c) 2022, Andreas Urke
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the Institute nor the names of its contributors
+ *    may be used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE INSTITUTE AND CONTRIBUTORS ``AS IS'' AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED.  IN NO EVENT SHALL THE INSTITUTE OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+ * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ *         Unit tests for the small PRNG library.
+ * \author
+ *         Andreas Urke
+ */
+
+#include "contiki.h"
+#include "lib/smallprng.h"
+
+#include "unit-test/unit-test.h"
+
+#include <stdio.h>
+#include <string.h>
+/*---------------------------------------------------------------------------*/
+PROCESS(test_smallprng_process, "Test small-PRNG library");
+AUTOSTART_PROCESSES(&test_smallprng_process);
+/*---------------------------------------------------------------------------*/
+#define RAND_BUF_SIZE 128
+/*---------------------------------------------------------------------------*/
+UNIT_TEST_REGISTER(output_sanity, "Basic sanity of the output");
+UNIT_TEST(output_sanity)
+{
+  UNIT_TEST_BEGIN();
+
+  smallprng_t smallprng1 = {0};
```

Comment:
This is immediately passed to `smallprng_init`, is the zero initialization required?

Reply:
I have the habit of doing this to avoid any future devs. using `smallprng1` in a different way without realizing it has not been initialized. And with the thinking that the compiler will avoid any performance penalty, or any penalty will be worth the reduced risk. But no strong opinion, happy to hear other thoughts.

Reply:
Sorry for the late response, but initializing the data will make the problem invisible for tools like Valgrind/Memory Sanitizer/etc, otherwise they would give warnings when data is used without being initialized.

Without entering the risk/reward analysis judgement, I want to mention that my experience is that only the most basic "sequential" double initializations will be reliably optimized away even by "modern" compilers (Clang/LLVM, ~2019). In this case, it seems unlikely the compiler will look into `smallprng_init` and deduce the zero-initialization isn't required.